Microsoft/WebTemplateStudio
# angular utilize form validation

This PR is to address #667 
Quote from this issue: "The Angular ListFormComponent should leverage angular forms and their built-in validation system, rather than delegating the validation to listeners of its @output() field. In this way, the component will only output a message once it gets validated by itself."

Also, utilized ng-bootstrap library for styling alter. Code Example can be found here.
https://ng-bootstrap.github.io/#/components/alert/examples

**How to test:**
1. generate an angular app with list page selected
2. start the app, add and delete list should work properly.
-  Without enter anything, the Submit button should be disabled
- When enter some chars and completely removed the chars, an error message should show up
![ng-form](https://user-images.githubusercontent.com/12874317/59811858-9a343680-92c0-11e9-9a73-a37ec7e361c9.PNG)


closes #667
